src/Plugin/KeyType/ApigeeAuthKeyType.php

```diff
@@ -0,0 +1,135 @@
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Plugin\KeyType;
+
+use Drupal\apigee_edge\OauthAuthentication;
+use Drupal\apigee_edge\Plugin\EdgeKeyTypeBase;
+use Drupal\apigee_edge\Plugin\EdgeKeyTypeInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\key\KeyInterface;
+use Http\Message\Authentication;
+use Http\Message\Authentication\BasicAuth;
+
+/**
+ * Key type for Apigee Edge authentication credentials.
+ *
+ * @KeyType(
+ *   id = "apigee_auth",
+ *   label = @Translation("Apigee Edge Authentication"),
+ *   description = @Translation("Key type to use for Apigee Edge authentication credentials."),
+ *   group = "apigee_edge",
+ *   key_value = {
+ *     "plugin" = "apigee_auth_input"
+ *   },
+ *   multivalue = {
+ *     "enabled" = true,
+ *     "fields" = {
+ *       "auth_type" = {
+ *         "label" = @Translation("Authentication type"),
+ *         "required" = true
+ *       },
+ *       "organization" = {
+ *         "label" = @Translation("Organization"),
+ *         "required" = true
+ *       },
+ *       "username" = {
+ *         "label" = @Translation("Username"),
+ *         "required" = true
+ *       },
+ *       "password" = {
+ *         "label" = @Translation("Password"),
+ *         "required" = true
+ *       },
+ *       "endpoint" = {
+ *         "label" = @Translation("Apigee Edge endpoint"),
+ *         "required" = false
+ *       },
+ *       "authorization_server" = {
+ *         "label" = @Translation("Authorization server"),
+ *         "required" = false
+ *       },
+ *       "client_id" = {
+ *         "label" = @Translation("Client ID"),
+ *         "required" = false
+ *       },
+ *       "client_secret" = {
+ *         "label" = @Translation("Client secret"),
+ *         "required" = false
+ *       }
+ *     }
+ *   }
+ * )
+ */
+class ApigeeAuthKeyType extends EdgeKeyTypeBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function generateKeyValue(array $configuration) {
+    return '[]';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateKeyValue(array $form, FormStateInterface $form_state, $key_value) {
+    if (empty($key_value)) {
+      return;
+    }
+
+    $value = $this->unserialize($key_value);
+    if ($value === NULL) {
+      $form_state->setError($form, $this->t('The key value does not contain valid JSON: @error', ['@error' => json_last_error_msg()]));
+      return;
+    }
+
+    foreach ($this->getPluginDefinition()['multivalue']['fields'] as $id => $field) {
+      if (isset($field['required']) && $field['required'] === FALSE) {
+        continue;
+      }
+
+      $error_element = $form['settings']['input_section']['key_input_settings'][$id] ?? $form;
+
+      /** @var \Drupal\Core\StringTranslation\TranslatableMarkup $field */
```

Comment:
Is this annotation actually valid and useful? I see only usages of array elements from the $field array below.